lib/sanitiser/strategy.rb

```diff
@@ -0,0 +1,18 @@
+module Sanitiser
+  class Strategy
+    class SanitisingError < StandardError; end
+
+    class << self
+      def call(input, sanitize_null_bytes: false)
```

Comment:
Sorry I don't understand how `sanitize_null_bytes` can ever be set to `true` ?

But can't we just pass in `sanitize_null_bytes: true` in the application config file, as it recommends [here](https://github.com/whitequark/rack-utf8_sanitizer?tab=readme-ov-file#sanitizing-null-bytes) and allow the default error and message to be raised?

Reply:
[sanitize_null_bytes](https://github.com/whitequark/rack-utf8_sanitizer/blob/f2bcf865432a1470c30de3519df780a5b39cf9a2/lib/rack/utf8_sanitizer.rb#L23) is an option which can be [potentially specified](https://github.com/whitequark/rack-utf8_sanitizer/tree/main?tab=readme-ov-file#sanitizing-null-bytes). I didn't specify it, so it won't be set, but it safer for the strategy to support it to be bullet proof against someone setting this option at some point in the future. The strategy is based on the original strategy from the gem which you can [see here](https://github.com/whitequark/rack-utf8_sanitizer/blob/f2bcf865432a1470c30de3519df780a5b39cf9a2/lib/rack/utf8_sanitizer.rb#L47). I just changed the exception raised.

Reply:
Why have you created this custom of strategy? Is it for the reason I've suggested [here](https://github.com/alphagov/government-frontend/pull/3371#pullrequestreview-2389192308) ?

Reply:
We need a custom error so we can safely ignore it. The default strategy raises e.g. EncodingError which is [one of a standard ruby exceptions](https://ruby-doc.org/3.3.5/EncodingError.html). We cannot ignore all EncodingErrors, because they can come from different places and for different reasons. So we should only ignore those that relate to UTF-8 encoding issues in query parameters (and as discussed earlier, in post data). We shouldn't ignore all EncodingErrors, as those might be related to some unrelated problems which we will miss, if we ignore all of them.